### PR TITLE
The error about the development of the GPU-acceleration soil mechanics.

### PR DESCRIPTION
**Pull request by LZUlis**
I'm tring on the GPU-accelerated soil mechanics.
I encountered an iussue.

The error is at acoustic_step_2nd_half.hpp, Line 218
It can be compiled normally, but an error occurs when running
And the ERORR is:

    UR CUDA ERROR:  
	Value:           700  
	Name:            CUDA_ERROR_ILLEGAL_ADDRESS  
	Description:     an illegal memory access was encountered  
	Function:        wait  
	Source Location: /tmp/tmp.7vgJ2wJCWQ/intel-llvm-mirror/sycl/plugins/unified_runtime/ur/adapters/cuda/event.cpp:140  
        terminate called after throwing an instance of 'sycl::_V1::runtime_error'  
        what():  Native API failed. Native API returns: -999 (Unknown PI error) -999 (Unknown PI error)  
        Aborted (core dumped) 

Detail desription can be found in acoustic_step_2nd_half.hpp, Line 218

### CODE REVIEW

**Comment by Xiangyu-Hu** [comment]
@LZUlis It would be better not change the readme file.

File: tests/2d_examples/2d_examples_ck/test_2d_dambreak_ck/dambreak_ck.cpp
Comment (Xiangyu-Hu) [inline tests/2d_examples/2d_examples_ck/test_2d_dambreak_ck/dambreak_ck.cpp:1]: Please also not change thus file.

**Comment by Xiangyu-Hu** [comment]
@LZUlis Another issue is that we do not use files with the same name.

File: src/shared/shared_ck/particle_dynamics/fluid_dynamics/acoustic_step_1st_half.h
Comment (Xiangyu-Hu) [inline src/shared/shared_ck/particle_dynamics/fluid_dynamics/acoustic_step_1st_half.h:1]: You will not change any files already in the library for new feature. You need to write new class.
Or you first generalize the old code without change any functions, then add new feature as an extension.

File: src/src_sycl/shared/particle_dynamics/plastic_functions.h
Comment (Xiangyu-Hu) [inline src/src_sycl/shared/particle_dynamics/plastic_functions.h:1]: We do not need new files includes sycl headers. What we need is simply  introduce computing kernel for old classes.

File: src/shared/shared_ck/particle_dynamics/fluid_dynamics/acoustic_step_1st_half.h
Comment (Xiangyu-Hu) [reply src/shared/shared_ck/particle_dynamics/fluid_dynamics/acoustic_step_1st_half.h:1]: > Thanks.
> 
> My original plan was to add functions to the original code first, and then organize all changes after achieve all functions.
> 
> I will follow this rule in the future.

No. That is not how a code development should be.

File: tests/2d_examples/2d_examples_ck/test_2d_dambreak_ck/dambreak_ck.cpp
Comment (LZUlis) [reply tests/2d_examples/2d_examples_ck/test_2d_dambreak_ck/dambreak_ck.cpp:1]: I will follow the guidelines you mentioned in the future

**Comment by Xiangyu-Hu** [comment]
@LZUlis You basically only need to rewrite the corresponding soil code (using computing kernel in share_ck) to reproduce the same results as before.

File: src/shared/shared_ck/particle_dynamics/fluid_dynamics/acoustic_step_1st_half.h
Comment (LZUlis) [reply src/shared/shared_ck/particle_dynamics/fluid_dynamics/acoustic_step_1st_half.h:1]: got and learn it

**Comment by LZUlis** [comment]
> @LZUlis You basically only need to rewrite the corresponding soil code (using computing kernel in share_ck) to reproduce the same results as before.

Yes, I agree, I have copied the original fluid ck intergration 1st and 2nd with new classes/functions, and I am currently modifying this code based on the exisitng soil mechanics code

**Comment by Xiangyu-Hu** [comment]
> > @LZUlis You basically only need to rewrite the corresponding soil code (using computing kernel in share_ck) to reproduce the same results as before.
> 
> Yes, I agree, I have copied the original fluid ck intergration 1st and 2nd with new classes/functions, and I am currently modifying this code based on the exisitng soil mechanics code

No. As the soil code does not modify fluid dynamics code, you need not do that. Also the main time integration code for fluid dynamics already works for GPU.

**Comment by Xiangyu-Hu** [comment]
@LZUlis you can check how I rewriting the liner kernel correction from pull request #701 .

**Comment by LZUlis** [comment]
> @LZUlis you can check how I rewriting the liner kernel correction from pull request #701 .

Got it

**Pull request closed without merging**